snap-build/snapcraft-snap-build.yaml

```diff
@@ -60,38 +74,54 @@ parts:
       '*': usr/share/doc/boot-firmware/
 
   u-boot-mtk:
+    after:
+      - test-keys
     plugin: nil
     source-type: git
     source: https://gitlab.com/mediatek/aiot/bsp/u-boot.git
     #source-branch: mtk-v2022.10
-    source-commit: 1d6626ab6e27b991c9f0159462d945d64c2aa9ec
+    source-commit: f7a807b03b17670f841b0c173c5a80c6d0d862fb
     source-depth: 1
     build-environment:
+      - SOURCE_TAG: rity-kirkstone-v24.1
       - CROSS_COMPILE: "${CRAFT_ARCH_TRIPLET}-"
       - U_BOOT_CONFIG: genio_1200_evk_defconfig
     override-pull: |
       craftctl default
-      SRC_URL="https://gitlab.com/mediatek/aiot/rity/meta-mediatek-bsp/-/raw/rity-kirkstone-v24.0/recipes-bsp/u-boot/files/"
+      SRC_URL="https://gitlab.com/mediatek/aiot/rity/meta-mediatek-bsp/-/raw/${SOURCE_TAG}/recipes-bsp/u-boot/files/"
       wget ${SRC_URL}/0001-Revert-cmd-pxe_utils-Check-fdtcontroladdr-in-label_b.patch
```

Comment:
wget should be added in `build-packages`

Reply:
Hi @Kxuan 
Based on https://documentation.ubuntu.com/snapcraft/stable/explanation/parts-lifecycle/
Pull. Downloads or otherwise retrieves the components needed to build the part.

Additional patches are components needed to build the part.
It seems to me that we should be able to use curl or wget in override-pull.
wget should be added in build-packages => Could you provide documentation for it?

For replacing wget with curl, I'll send another commit for it.